Training24/ClassicLibrary.c

```diff
@@ -1,12 +1,34 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
+// Kalaimagal V P
 // ------------------------------------------------------------------------------------------------
-// Classiclibrary.c
-// Program on main branch.
+// ClassicLibrary.c
+// Program on A4 branch.
+// 1. Palindrome is a word that reads the same forward or backward. Example: Civic, Madam, Bib, Nun, Level and so on. 
+//    Your program must take the word as input and display the result if it is a palindrome or not. 
+//    For complexity, you can also use phrases instead of words (E.g., Was it a car or a cat I saw? I did, did I? Don�t nod)
+// 2. Write a program that reverses a given integer (as defined in C) as explained below and check if it is a palindrome.
 // ------------------------------------------------------------------------------------------------
-char input[] = "Hello World!"; // Global character array
+#include <stdbool.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <string.h>
 
-char* GetString () {
-   return input; // Return the global greeting
+bool IsPalindrome (const char* str) {
+   size_t left = 0, right = strlen (str) - 1;
+   while (left < right) if (str[left++] != str[right--]) return false;  // Return false if characters at 'left' and 'right' differ.
+   return true;                                                         // All characters matched; it's a palindrome.
+}
+
+int ReverseNumber (int num) {
+   int reversed = 0,negative = (num < 0) ? 1 : 0;

```

Comment:
Here, you are assigning negative a boolean value. The expression 'num < 0' also returns a boolean value, which is why 
negative = num < 0 is sufficient.
```suggestion
   int reversed = 0, negative = num < 0;

```